
<launch>
  <arg name="serial_no_camera1"    			default="243522073159"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera2"    			default="242222070936"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="camera1"              			default="left_camera"/>		<!-- Note: Replace with camera name -->
  <arg name="camera2"              			default="right_camera"/>		<!-- Note: Replace with camera name -->
  <arg name="tf_prefix_camera1"         default="$(arg camera1)"/>
  <arg name="tf_prefix_camera2"         default="$(arg camera2)"/>
  <arg name="initial_reset"             default="false"/>
  <arg name="reconnect_timeout"         default="6.0"/>

  <group ns="$(arg camera1)"> 
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="serial_no"             value="$(arg serial_no_camera1)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_camera1)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="reconnect_timeout"     value="$(arg reconnect_timeout)"/>
      <arg name="enable_pointcloud" value="true"/>
      <arg name="enable_sync"       value="true"/>
    </include>
  </group>

  <group ns="$(arg camera2)">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="serial_no"             value="$(arg serial_no_camera2)"/>
      <arg name="tf_prefix"		          value="$(arg tf_prefix_camera2)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="reconnect_timeout"     value="$(arg reconnect_timeout)"/>
      <arg name="enable_pointcloud" value="true"/>
      <arg name="enable_sync"       value="true"/>
    </include>
  </group>
  
  <include file="$(find box_detection)/launch/robot.launch"/>

  <!-- <node name="aruco" pkg="box_detection" type="detect.py" output="screen"/> -->


  <node name="left_camera_broadcast" pkg="box_detection" type="publish_camera.py" output="screen" required="true">
        <param name="input_file" value="/home/riot/kinova_gen3_lite/src/box_detection/cam_poses/camera_pose_left.txt"/>
        <param name="parent_frame" value="base_link"/>
        <param name="child_frame" value="left_camera_link"/>
  </node>

  <node name="right_camera_broadcast" pkg="box_detection" type="publish_camera.py" output="screen" required="true">
      <param name="input_file" value="/home/riot/kinova_gen3_lite/src/box_detection/cam_poses/camera_pose_right.txt"/>
      <param name="parent_frame" value="base_link"/>
      <param name="child_frame" value="right_camera_link"/>
  </node>
</launch>
